src/main/java/uk/gov/pay/products/service/ProductFinder.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.pay.products.service;
+
+import com.google.inject.Inject;
+import uk.gov.pay.products.model.Product;
+import uk.gov.pay.products.persistence.dao.ProductDao;
+
+import java.util.Optional;
+
+public class ProductFinder {
+    private final ProductDao productDao;
+    private final LinksDecorator linksDecorator;
+
+    @Inject
+    public ProductFinder(ProductDao productDao, LinksDecorator linksDecorator) {
```

Comment:
We need a unit test for these service class (mockito based) have a look at adminusers
Also i think ProductCreator missing unit tests as well